osx/make-installer.sh

```diff
@@ -108,80 +108,52 @@ if [ ! -d "${app}/Contents/Frameworks/Sparkle.framework" ]; then
 
 # Bundle in dynamically loaded libraries
-cp "${HOME}/.luarocks/lib/lua/5.1/lfs.so" "${app}/Contents/MacOS"
-
-cp "${HOME}/.luarocks/lib/lua/5.1/rex_pcre.so" "${app}/Contents/MacOS"
+# These will be manually fixed up because macdeployqtfix is not really designed to handle individual libraries.
+echo "Bundling dynamic libraries"
+cp -v "${HOME}/.luarocks/lib/lua/5.1/lfs.so" "${app}/Contents/MacOS"
+cp -v "${HOME}/.luarocks/lib/lua/5.1/rex_pcre.so" "${app}/Contents/MacOS"
 # rex_pcre has to be adjusted to load libpcre from the same location
-python macdeployqtfix.py "${app}/Contents/MacOS/rex_pcre.so" "${QT_DIR}"
+install_name_tool -change "${HOMEBREW_PREFIX}/opt/pcre/lib/libpcre.1.dylib" "@executable_path/../Frameworks/libpcre.1.dylib" "${app}/Contents/MacOS/rex_pcre.so"
 
 cp -r "${HOME}/.luarocks/lib/lua/5.1/luasql" "${app}/Contents/MacOS"
-cp /usr/local/opt/sqlite/lib/libsqlite3.0.dylib  "${app}/Contents/Frameworks/"
-# sqlite3 has to be adjusted to load libsqlite from the same location
-python macdeployqtfix.py "${app}/Contents/Frameworks/libsqlite3.0.dylib" "${QT_DIR}"
+cp -v ${HOMEBREW_PREFIX}/opt/sqlite/lib/libsqlite3.0.dylib  "${app}/Contents/Frameworks/"
+install_name_tool -id  "@executable_path/../Frameworks/libsqlite3.0.dylib" "${app}/Contents/Frameworks/libsqlite3.0.dylib" 
 # need to adjust sqlite3.lua manually as it is a level lower than expected...
-install_name_tool -change "/usr/local/opt/sqlite/lib/libsqlite3.0.dylib" "@executable_path/../../Frameworks/libsqlite3.0.dylib" "${app}/Contents/MacOS/luasql/sqlite3.so"
+install_name_tool -change "${HOMEBREW_PREFIX}/opt/sqlite/lib/libsqlite3.0.dylib" "@executable_path/../../Frameworks/libsqlite3.0.dylib" "${app}/Contents/MacOS/luasql/sqlite3.so"
 
-cp "${HOME}/.luarocks/lib/lua/5.1/lua-utf8.so" "${app}/Contents/MacOS"
+cp -v "${HOME}/.luarocks/lib/lua/5.1/lua-utf8.so" "${app}/Contents/MacOS"
 
 # The lua-zip rock:
 # Also need to adjust the zip.so manually so that it can be at a level down from
 # the executable:
 mkdir "${app}/Contents/MacOS/brimworks"
-cp "${HOME}/.luarocks/lib/lua/5.1/brimworks/zip.so" "${app}/Contents/MacOS/brimworks"
-# Special case - libzip.5.dylib in Github Actions is located in this path
-if [ -n "$GITHUB_REPOSITORY" ] ; then
-  mkdir -p "/usr/local/lib/libzip.5.dylib.framework"
-  cp "/usr/local/lib/libzip.5.dylib" "/usr/local/lib/libzip.5.dylib.framework/libzip.5.dylib"
-fi
-python macdeployqtfix.py "${app}/Contents/MacOS/brimworks/zip.so" "/usr/local"
-
-# Also fix up a separate (does it have to be?) copy of libzip which has a
-# dependency on libzstd (and liblzma) and copy them into
-# "${app}/Contents/Frameworks" - which is a sort of standard placement in App
-# bundles - to enable things to keep working when minor or patch version numbers
-# change it is necessary to use some wildcards, these may actually target a
-# symbolic link but when they are copied the linked to file will be what is
-# copied:
```

Comment:
The diff is a little hard to read, but the net issue is that I found that these dylibs were already being bundled by macdeployqt and I couldn't see another reason to copy them over again and update their library paths, so I removed this part. I'm sure it was added in for a reason though, so let me know.

Reply:
Not sure if this is relevant here but IIRC I would note that the lua sub-system treats '.' other that a final one preceding an extension as a directory separator when parsing a pathAndFileName for a Lua package to load.

Also the **[`lua-zip`](https://github.com/brimworks/lua-zip)** Luarock is a different one to the **[`luazip`](https://mpeterv.github.io/luazip/)** one (it DOES NOT have a dependency on the `libzzip` or `zziplib` or `zzip` library - the name of that one is a bit variable but it has two 'z'!) and the first one, the "Brimworks" one, is expected to be put in a subdirectory (by it's author who also made the `lua-yajl` library/Luarock!) so it does not clash with the other version or the `zziplib` library - see the README.md for the details...

Reply:
A thing I discovered:
The brimworks one (lua-zip) is currently being put in the right place.
luazip (no space) is not being installed at all (in the current release, and in this PR)
However, the module load in has this code:

```
    bool loaded = loadLuaModule(modLoadMessageQueue, QLatin1String("brimworks.zip"), QString(), qsl("lua-zip"), qsl("zip"));
    if (!loaded) {
        loadLuaModule(modLoadMessageQueue, QLatin1String("zip"), QString(), qsl("luazip"));
    }
```

Meaning that it will try to load luazip if for some reason the brimworks one is unavailable. This fallback is currently broken because luazip is never bundled.

Question is: do we care enough about that fallback behavior to bundle luazip? In what cases would this fallback ever happen, given that we bundle brimworks / lua-zip?

Reply:
We *used* to bundle `luazip` but that dependency on `zziplib` borked things for us a while back.

Reply:
We don't need the fallback I don't think. It hasn't been necessary.

Reply:
> Not sure if this is relevant here but IIRC I would note that the lua sub-system treats '.' other that a final one preceding an extension as a directory separator when parsing a pathAndFileName for a Lua package to load.

I should have said: "the lua sub-system treats '.' as a directory separator when parsing a name for a Lua package to `require(...)`" - but you have understood it!

Reply:
Ok, seems like status quo is good so will not make any changes to this part.